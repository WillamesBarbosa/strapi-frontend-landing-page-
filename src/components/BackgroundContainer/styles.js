import styled, { css } from 'styled-components';

const handleChangeBackgroundColor = (theme) => css`
  background-color: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, backgroundColor }) => css`
    background-color: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${backgroundColor && handleChangeBackgroundColor(theme)}
    min-height: 100vh;
    padding: 5rem;
  `}
`;
