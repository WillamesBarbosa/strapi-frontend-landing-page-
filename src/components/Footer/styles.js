import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.size.small};
    }
    text-align: center;
  `}
`;
