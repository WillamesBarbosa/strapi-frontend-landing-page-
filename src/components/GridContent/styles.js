import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.font.spacings.xbig} 0;
  `}
`;
