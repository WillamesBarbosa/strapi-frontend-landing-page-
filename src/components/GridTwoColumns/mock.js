export default {
  title: 'Title Text',
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Sit deleniti ullam magnam! Nam unde porro voluptatum accusantium dicta amet quaerat illum laborum ab animi. Rerum excepturi aperiam aliquid sint velit?`,
  srcImg: 'assets/javascript.svg',
};
