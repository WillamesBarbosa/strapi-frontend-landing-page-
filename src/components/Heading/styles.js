import styled, { css } from 'styled-components';

const titleSize = {
  xxsmall: (theme) =>
    css`
      font-size: ${theme.font.size.xxsmall};
    `,
  xsmall: (theme) =>
    css`
      font-size: ${theme.font.size.xsmall};
    `,
  small: (theme) =>
    css`
      font-size: ${theme.font.size.small};
    `,
  medium: (theme) =>
    css`
      font-size: ${theme.font.size.medium};
    `,
  big: (theme) =>
    css`
      font-size: ${theme.font.size.big};
    `,
  xbig: (theme) =>
    css`
      font-size: ${theme.font.size.xbig};
    `,
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const mediaFont = (theme) => css`
  @media ${theme.mediaQuerie.mediumQuerie} {
    font-size: ${theme.font.size.medium};
  }
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)}
    ${titleCase(uppercase)}
    ${mediaFont(theme)}
  `}
`;
