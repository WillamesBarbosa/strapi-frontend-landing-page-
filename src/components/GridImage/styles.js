import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    margin: auto;
    width: 85%;
    ${TextComponent} {
      margin-bottom: ${theme.font.spacings.xbig};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.font.spacings.big};
  `}
`;

export const GridElement = styled.div`
  ${() => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${() => css`
    width: 100%;

    &:hover {
      transform: scale(1.2);
    }
  `}
`;
