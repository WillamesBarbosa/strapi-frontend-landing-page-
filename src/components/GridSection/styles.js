import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.font.spacings.xbig};
      margin-top: ${theme.font.spacings.xbig};
      font-size: ${theme.font.spacings.small};

      & > ${HeadingContainer} {
        font-size: ${theme.font.size.small};
      }
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.font.spacings.big};
    overflow: hidden;
    margin: auto;
    width: 80%;
    @media ${theme.mediaQuerie.mediumQuerie} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      font-size: ${theme.font.size.medium};
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: 7rem;
      top: -3rem;
      left: -5rem;
      transform: rotate(5deg);
    }
  `}
`;
