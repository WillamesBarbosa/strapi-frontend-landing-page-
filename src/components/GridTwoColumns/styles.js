import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.font.spacings.big};
    padding: ${theme.font.spacings.big};
    width: 85%;
    margin: auto;

    @media ${theme.mediaQuerie.mediumQuerie} {
      grid-template-columns: 1fr;
    }

    & ${Title} {
      margin-bottom: ${theme.font.spacings.big};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.mediaQuerie.mediumQuerie} {
      margin-bottom: ${theme.font.spacings.big};
    }
  `}
`;

export const ImageContainer = styled.div``;

export const Image = styled.img``;
