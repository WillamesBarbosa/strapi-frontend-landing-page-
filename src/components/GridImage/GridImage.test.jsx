import { renderTheme } from '../../styles/render-theme';
import { GridImage } from '.';
import mock from './mock';
import { screen } from '@testing-library/react';

describe('<GridImage />', () => {
  it('should match snapshot', () => {
    const { container } = renderTheme(<GridImage {...mock} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render GridImage dark', () => {
    renderTheme(<GridImage {...mock} background={true} />);
    expect(screen.getByAltText('Teste 1')).toBeInTheDocument();
  });

  it('should render GridImage undefined background', () => {
    renderTheme(<GridImage {...mock} background={undefined} />);
    expect(screen.getByAltText('Teste 1')).toBeInTheDocument();
  });

  it('should render GridImage light', () => {
    renderTheme(<GridImage {...mock} background={false} />);
    expect(screen.getByAltText('Teste 1')).toBeInTheDocument();
  });
});
