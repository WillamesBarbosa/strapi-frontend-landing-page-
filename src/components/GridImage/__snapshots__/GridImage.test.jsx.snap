// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridImage /> should match snapshot 1`] = `
.c4 {
  font-size: 3.2rem;
}

.c1 .c3 {
  margin-bottom: 4.8rem;
}

.c5 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(280px,1fr));
  gap: 4.0rem;
}

.c6 {
  overflow: hidden;
}

.c7 {
  width: 100%;
}

.c7:hover {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c0 {
  background-color: #fff;
  color: #0A1128;
  min-height: 100vh;
  padding: 5rem;
}

.c2 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: none;
}

@media (max-width:768px) {
  .c2 {
    font-size: 3.2rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
    >
      My grid
    </h2>
    <div
      class="c3 c4"
    >
      Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde
    </div>
    <div
      class="c5"
    >
      <div
        class="c6"
      >
        <img
          alt="Teste 1"
          class="c7"
          src="https://source.unsplash.com/random/800x800?r=1"
        />
      </div>
      <div
        class="c6"
      >
        <img
          alt="Teste 2"
          class="c7"
          src="https://source.unsplash.com/random/800x800?r=2"
        />
      </div>
      <div
        class="c6"
      >
        <img
          alt="Teste 3"
          class="c7"
          src="https://source.unsplash.com/random/800x800?r=3"
        />
      </div>
      <div
        class="c6"
      >
        <img
          alt="Teste 4"
          class="c7"
          src="https://source.unsplash.com/random/800x800?r=4"
        />
      </div>
      <div
        class="c6"
      >
        <img
          alt="Teste 5"
          class="c7"
          src="https://source.unsplash.com/random/800x800?r=5"
        />
      </div>
      <div
        class="c6"
      >
        <img
          alt="Teste 6"
          class="c7"
          src="https://source.unsplash.com/random/800x800?r=6"
        />
      </div>
    </div>
  </div>
</div>
`;
